File: app/src/main/java/com/example/disintegrator/MainActivity.java
package com.example.disintegrator;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button scanButton = findViewById(R.id.ScanButton);
        scanButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                goToScanView(view);
            }
        });
    }

    private void goToScanView(View view) {
        System.err.println("Start goToScanView");
        Intent scanView = new Intent(view.getContext(), ScannerActivityView.class);
        view.getContext().startActivity(scanView);
    }
}

File: app/src/main/java/com/example/disintegrator/MainActivity.kt
package com.example.disintegrator

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val scanButton = findViewById<Button>(R.id.ScanButton)
        scanButton.setOnClickListener { view -> goToScanView(view) }
    }

    private fun goToScanView(view: View) {
        System.err.println("Start goToScanView")
        val scanView = Intent(view.context, ScannerActivityView::class.java)
        view.context.startActivity(scanView)
    }
}